### PR TITLE
feat(Tag): add Tag component

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview

Adds new `Tag` component

### PR Checklist

- [x] Related to designs: [Tags](https://www.figma.com/file/a0xaVwpkrXpOReMJ3ArqCe/Badges?node-id=1%3A40)
- [x] Related to JIRA ticket: WEB-2004
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

### Testing Instructions
* Check out the Tags story 
* Make sure the interactive example works
* Play around w/ the Code Sandbox

#### PR Links and Envs
none! it's new ✨

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.137-alpha.8aefa4.0<br/>@codecademy/gamut-labs@31.15.5-alpha.8aefa4.0<br/>@codecademy/gamut@47.4.1-alpha.8aefa4.0<br/>@codecademy/monaco@0.6.49-alpha.8aefa4.0<br/>@codecademy/styleguide@57.7.5-alpha.8aefa4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://629fc7a5e9d504008b96579d--gamut-preview.netlify.app>https://629fc7a5e9d504008b96579d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/629fc7a5e9d504008b96579d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.153-alpha.f62c99.0<br/>@codecademy/gamut-labs@32.4.1-alpha.f62c99.0<br/>@codecademy/gamut@48.0.7-alpha.f62c99.0<br/>@codecademy/monaco@0.6.56-alpha.f62c99.0<br/>@codecademy/styleguide@58.1.8-alpha.f62c99.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62c5bb9d895a890059ef8f7d--gamut-preview.netlify.app>https://62c5bb9d895a890059ef8f7d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62c5bb9d895a890059ef8f7d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.155-alpha.6e775f.0<br/>@codecademy/gamut-labs@32.5.2-alpha.6e775f.0<br/>@codecademy/gamut@48.0.7-alpha.6e775f.0<br/>@codecademy/monaco@0.6.56-alpha.6e775f.0<br/>@codecademy/styleguide@58.1.10-alpha.6e775f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62c72843a62d6627d3a6e57c--gamut-preview.netlify.app>https://62c72843a62d6627d3a6e57c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62c72843a62d6627d3a6e57c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.155-alpha.adb790.0<br/>@codecademy/gamut-labs@32.5.2-alpha.adb790.0<br/>@codecademy/gamut@48.0.7-alpha.adb790.0<br/>@codecademy/monaco@0.6.56-alpha.adb790.0<br/>@codecademy/styleguide@58.1.10-alpha.adb790.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62c734613074b5316ddd19d6--gamut-preview.netlify.app>https://62c734613074b5316ddd19d6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62c734613074b5316ddd19d6>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.155-alpha.f5c3a3.0<br/>@codecademy/gamut-labs@32.5.2-alpha.f5c3a3.0<br/>@codecademy/gamut@48.0.7-alpha.f5c3a3.0<br/>@codecademy/monaco@0.6.56-alpha.f5c3a3.0<br/>@codecademy/styleguide@58.1.10-alpha.f5c3a3.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.156-alpha.448fb7.0<br/>@codecademy/gamut-labs@32.6.1-alpha.448fb7.0<br/>@codecademy/gamut@48.0.7-alpha.448fb7.0<br/>@codecademy/monaco@0.6.56-alpha.448fb7.0<br/>@codecademy/styleguide@58.1.11-alpha.448fb7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62cc6ad98c08450094dbf237--gamut-preview.netlify.app>https://62cc6ad98c08450094dbf237--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62cc6ad98c08450094dbf237>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62cc7e4feececa0e5f5159e1--gamut-preview.netlify.app>https://62cc7e4feececa0e5f5159e1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62cc7e4feececa0e5f5159e1>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.156-alpha.28333f.0<br/>@codecademy/gamut-labs@32.6.1-alpha.28333f.0<br/>@codecademy/gamut@48.0.7-alpha.28333f.0<br/>@codecademy/monaco@0.6.56-alpha.28333f.0<br/>@codecademy/styleguide@58.1.11-alpha.28333f.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.156-alpha.a872c2.0<br/>@codecademy/gamut-labs@32.6.1-alpha.a872c2.0<br/>@codecademy/gamut@48.0.7-alpha.a872c2.0<br/>@codecademy/monaco@0.6.56-alpha.a872c2.0<br/>@codecademy/styleguide@58.1.11-alpha.a872c2.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62cc96846efc660ebdf91851--gamut-preview.netlify.app>https://62cc96846efc660ebdf91851--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62cc96846efc660ebdf91851>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.157-alpha.85f95f.0<br/>@codecademy/gamut-labs@32.6.2-alpha.85f95f.0<br/>@codecademy/gamut@48.0.7-alpha.85f95f.0<br/>@codecademy/monaco@0.6.56-alpha.85f95f.0<br/>@codecademy/styleguide@58.1.12-alpha.85f95f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62cf194e9951cf28af2841e9--gamut-preview.netlify.app>https://62cf194e9951cf28af2841e9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62cf194e9951cf28af2841e9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.158-alpha.181f88.0<br/>@codecademy/gamut-labs@32.7.1-alpha.181f88.0<br/>@codecademy/gamut@48.0.7-alpha.181f88.0<br/>@codecademy/monaco@0.6.56-alpha.181f88.0<br/>@codecademy/styleguide@58.2.1-alpha.181f88.0<br/></pre>

File: packages/styleguide/stories/Atoms/Tag.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        <Tag>I am long tag yes I am much am a very long tag</Tag>
+      </Box>
+    )}
+  </Story>
+</Canvas>
+
+## Dismissable Tags Example
+
+<Canvas>
+  <Story name="Working Example">{() => <TagExample />}</Story>
```
Comment (aresnik11) [inline packages/styleguide/stories/Atoms/Tag.stories.mdx:61]: i love this! makes it very easy to test out

**Review by aresnik11 — APPROVED** [review]
Have a few nits and questions but overall this is so cool and works great!!

Seeing in the props table that `variant` can be `false`, how/when does that happen?
Screen Shot 2022-07-14 at 4 00 35 PM

File: packages/gamut/src/Tag/index.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          px={8}
+          truncate="ellipsis"
+          truncateLines={1}
+        >
+          {children}
+        </Text>
+        {!readonly && (
+          <DeleteButton
+            aria-label={`Delete ${children} Tag`}
+            onClick={onDismiss || undefined}
```
[the remark this answers is not in the document]
Comment (dreamwasp) [reply packages/gamut/src/Tag/index.tsx:42]: yeah definitely -- i'll swap all of the verbiage over to `delete`!

File: packages/styleguide/stories/Atoms/Tag.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  args={{ children: 'Tag' }}
+/>
+
+## Usage Guidelines
+
+Tags are compact, dynamic elements that support user inputs, selections, and actions. Tags can either be dismissable individually, or read-only when paired with other controls. Use Tags in designs like Filters.
+
+## Variants
+
+Tags have a ColorModed `default` variant and a `grey` variant as well as a `readonly` variant
```
[the remark this answers is not in the document]
Comment (dreamwasp) [reply packages/styleguide/stories/Atoms/Tag.stories.mdx:30]: definitely can make it more clear -- ill reword!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.158-alpha.a2adef.0<br/>@codecademy/gamut-labs@32.7.1-alpha.a2adef.0<br/>@codecademy/gamut@48.0.7-alpha.a2adef.0<br/>@codecademy/monaco@0.6.56-alpha.a2adef.0<br/>@codecademy/styleguide@58.2.1-alpha.a2adef.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62d570dd6fdb6000615f9ba5--gamut-preview.netlify.app>https://62d570dd6fdb6000615f9ba5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62d570dd6fdb6000615f9ba5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.160-alpha.c15a04.0<br/>@codecademy/gamut-labs@32.7.3-alpha.c15a04.0<br/>@codecademy/gamut@48.1.1-alpha.c15a04.0<br/>@codecademy/monaco@0.6.57-alpha.c15a04.0<br/>@codecademy/styleguide@58.3.1-alpha.c15a04.0<br/></pre>

**Pull request merged**